edb/server/compiler/compiler.py

```diff
@@ -107,9 +110,9 @@ class CompileContext:
     bootstrap_mode: bool = False
     internal_schema_mode: bool = False
     standalone_mode: bool = False
+    system_config: immutables.Map = EMPTY_MAP
```

Comment:
Why is this here and not in CompilerConnectionState?

Reply:
System config isn't transactional, so it seemed unnecessary to do all the pointless transaction tracking for it in `CompilerConnectionState`.  That said, it might still worth it for consistency, I guess.